### DISCUSSION TITLE
Strongswan confuses the credentials in the ipsec.secret file with those in the secret section of the swanctl.conf file for the psk and eap methods

### DISCUSSION THREAD

**Discussion by sakscams** [strongswan/strongswan#644] [Q&A] [answered]
I have two configurations: one with ipsec.conf and the other with swanctl.conf.
The authentication information is not the same for the two configurations.
I disabled the starter service (ipsec.conf) to use the swanctl service. 
Now when I try to authenticate my VPN client with the PSK method, Strongswan accepts both secrets: the one defined in the secret section of the swanctl.conf file but also the one defined in ipsec.secrets although the latter is disabled.
it is the same for the EAP-mschap method, but for the latter, strongswan validates the authentication of the VPN client for the secret in ipsec.secrets but refuses the swanctl.conf one.

how to solve this problem so that Strongswan does not read in ipsec.secrets to authenticate the VPN client but only in swanctl.conf.

I am currently using strongSwan 5.9.1

**Comment by tobiasbrunner** [comment, marked as answer]
If you don't want the `ipsec.secrets` file to be read, [disable](https://wiki.strongswan.org/projects/strongswan/wiki/PluginLoad) the _stroke_ plugin. Or just delete the `ipsec.secrets` file or comment/remove the entries in it.

**Reply by sakscams** [reply]
it works, thanks !
